table/block_based/block_based_table_builder.cc

```diff
@@ -451,6 +461,105 @@ struct BlockBasedTableBuilder::Rep {
   ~Rep() {}
 };
 
+class BlockBasedTableBuilder::ParallelCompressionRep {
+ public:
+  // Keys is a wrapper of vector of strings avoiding
+  // releasing string memories during vector clear()
+  // in order to save memory allocation overhead
+  class Keys {
+   public:
+    Keys() : keys(kKeysInitSize), size(0) {}
+    void PushBack(const Slice& key) {
+      if (size == keys.size()) {
+        keys.emplace_back(key.data(), key.size());
+      } else {
+        keys[size].assign(key.data(), key.size());
+      }
+      size++;
+    }
+    void SwapAssign(std::vector<std::string>& keys_) {
+      size = keys_.size();
+      std::swap(keys, keys_);
+    }
+    void Clear() { size = 0; }
+    size_t Size() { return size; }
+    std::string& Back() { return keys[size - 1]; }
+    std::string& operator[](size_t idx) {
+      assert(idx < size);
+      return keys[idx];
+    }
+
+   private:
+    const size_t kKeysInitSize = 32;
+    std::vector<std::string> keys;
+    size_t size;
+  };
+  Keys* keys_ptr;
+
+  class BlockRepSlot;
+
+  // BlockRep instances are fetched from and recycled to
+  // block_rep_pool during parallel compression
+  struct BlockRep {
+    Slice contents;
+    std::string* data_ptr;
+    std::string* compressed_data_ptr;
+    CompressionType compression_type;
+    std::string* first_key_in_next_block_ptr;
+    Keys* keys_ptr;
+    BlockRepSlot* slot_ptr;
```

Comment:
I don't think we need `ptr` in those variable names. It's clear they are pointers because their types are pointers.

Reply:
Fixed.